packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -153,8 +161,54 @@
       };
     }
 
-    protected mounted(): void {
-      this.$x.emit('QueryPreviewRequestChanged', this.queryPreviewRequest);
+    /**
+     * The debounce method to trigger the request after the debounceTimeMs defined.
+     *
+     * @returns The search request object.
+     * @internal
+     */
+    protected get emitQueryPreviewRequestChanged(): DebouncedFunction<[SearchRequest]> {
+      return debounce(request => {
+        this.$x.emit('QueryPreviewRequestChanged', request);
+      }, this.debounceTimeMs);
+    }
+
+    /**
+     * Initialises watcher to emit debounced requests, and first value for the requests.
+     *
+     * @internal
+     */
+    protected created(): void {
+      this.$watch(
+        () => this.queryPreviewRequest,
+        request => this.emitQueryPreviewRequestChanged(request)
+      );
+      this.emitQueryPreviewRequestChanged(this.queryPreviewRequest);
+    }
+
+    /**
+     * Cancels the (remaining) requests when the component is destroyed
+     * via the debounce.cancel() method.
```

Comment:
Can we change this line to wrap `debounce.cancel()` as code for the docs? The IDE doesn't like it.
```suggestion
     * via the `debounce.cancel()` method.
```